src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,175 @@
+            startTime: deployResultTime.toISOString(),
+            endTime: new Date().toISOString()
+        };
+        await signRequest(requestOptions, node.root.credentials);
+
+        let detectorErrorMessage: string | undefined;
+        // wait 10 minutes for a response from the detector
+        const detectorTimeoutMs: number = Date.now() + 20 * 60 * 1000;
+        do {
+            if (tokenSource.token.isCancellationRequested) {
+                // the user cancelled the check by deploying again
+                throw new UserCancelledError();
+            }
+
+            if (Date.now() > detectorTimeoutMs) {
+                const noIssuesFound: string = `No critical issues found for web app "${node.root.client.siteName}".`;
+                ext.outputChannel.appendLine(noIssuesFound);
+                return undefined;
+            }
+            // update the new end time to be now
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(requestOptions, deployResultTime, node.root.client.fullName);
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+        const portalDeeplink = `${node.root.environment.portalUrl}/?websitesextension_ext=asd.featurePath%3Ddetectors%2F${detectorId}#@microsoft.onmicrosoft.com/resource/${node.root.client.id}/troubleshoot`;
+        await openUrl(portalDeeplink);
+        context.telemetry.properties.didClick = 'true';
+    });
+}
+
+async function signRequest(req: WebResource, cred: ServiceClientCredentials): Promise<void> {
+    await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
+        cred.signRequest(req, (err: Error | undefined) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+async function getLinuxDetectorError(requestOptions: WebResource, deployResultTime: Date, siteName: string): Promise<string | undefined> {
+    const requestPromise = <(options: RequestOptions | string | URL) => Promise<IncomingMessage>><Function>requestP;
+    const detectorResponse: string = <string><unknown>(await requestPromise(requestOptions));
+    const responseJson: detectorResponseJSON = <detectorResponseJSON>JSON.parse(detectorResponse);
+
+    const datasets: detectorDataset[] = responseJson.properties.dataset;
+    const vsCodeIntegration: string = 'Latest time seen by detector. To be used in VSCode integration.';
+    const timestampTable: detectorTable | undefined = findTableByRowValue(datasets, vsCodeIntegration);
+    if (!timestampTable) {
+        // if we can't find the timestamp, exit and try again
+        return undefined;
+    }
+
+    const timestamp = new Date(getValuesByColumnName(timestampTable, ColumnName.value)[0]);
```

Comment:
Add at least one unit test that validates essentially all of `getLinuxDetectorError` except the request part. Aka given `datasets`, it will return these errors and these failure counts